### PR TITLE
👽️ Migration to Cairo 2

### LINKED ISSUE #5: [Migration] Migrate cairo-migrator Smart Contract from Cairo 0.10 to Cairo 1
# Description

The cairo-migrator is writing in Cairo 0.10 and it is working. 

# Tasks

The Tasks are : 
- [ ] Migrate the `src/migrator` from Cairo 0.10 to Cairo 1 (preset, constants, library)
- [ ] Migrate the tests
- [ ] Add tests to migrate a Smart Contract from Cairo 0.10 to Cairo 1

### PR DESCRIPTION
**Pull request by bal7hazar**
Closes #5

### CODE REVIEW

File: src/components/migrate/module.cairo [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        fn source_address(self: @ContractState) -> starknet::ContractAddress {
+            self._migrate_source_address.read()
+        }
+
+        fn target_address(self: @ContractState) -> starknet::ContractAddress {
+            self._migrate_target_address.read()
+        }
+
+        fn slot(self: @ContractState) -> u256 {
+            self._migrate_slot.read()
```
Comment (tekkac) [inline src/components/migrate/module.cairo:62]: Do we need to check that token_ids are in the right slot?

File: src/tests/test_migrator.cairo [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+fn test_migrator_migrate() {
+    // [Setup]
+    let (signers, contracts) = setup();
+    // [Assert] Approve and Migrate
+    set_contract_address(signers.anyone);
+    let erc721 = IERC721Dispatcher { contract_address: contracts.source };
+    erc721.set_approval_for_all(contracts.migrator, true);
+    let migrator = IMigrateDispatcher { contract_address: contracts.migrator };
+    let token_id = migrator.migrate(array![ONE, TWO, THREE].span());
+    // [Assert] Value
```
Comment (tekkac) [inline src/tests/test_migrator.cairo:99]: Do we need those tests?:
- add tests with 0, 1 and 128 tokens?
- Check tokens not in same slots fails
- Check tokens not of owner fails

**Review by tekkac — APPROVED** [review]
LGTM :ok_hand:

File: src/components/migrate/module.cairo (hunk 1 above)
Comment (bal7hazar) [reply src/components/migrate/module.cairo:62]: token_ids are related to 721, there is no slot.

File: src/tests/test_migrator.cairo (hunk 2 above)
Comment (bal7hazar) [reply src/tests/test_migrator.cairo:99]: add tests with 0, 1 and 128 tokens? :+1: (done with 100)
Check tokens not in same slots fails :-1: 
Check tokens not of owner fails :+1:

**Review by tekkac — APPROVED** [review]
:muscle:

**Pull request merged**